go.mod

```diff
@@ -84,7 +84,6 @@ require (
 	github.com/jackc/pgservicefile v0.0.0-20221227161230-091c0ba34f0a // indirect
 	github.com/jackc/pgtype v1.14.0 // indirect
 	github.com/jackc/pgx/v4 v4.18.2 // indirect
-	github.com/jackc/puddle v1.3.0 // indirect
```

Comment:
can squash this commit into previous one which does the go.mod update :) but non-blocking

Reply:
The reason i separated these into different commits is that the changes between those 2 commits are actually unrelated from each other, as project hadn't run `go mod tidy` for the previously merged PR. So even though they're both changing the `go.mod` file, neither of the commits are actually related :).

I should have clarified this, and In hindsight maybe it'd make sense to include the `go mod tidy` commit as the first commit in the PR for clarification. But choosing to merge this as is to avoid another CI run.